relay-event-schema/src/protocol/security_report.rs

```diff
@@ -442,6 +481,26 @@ struct CspReportRaw {
     csp_report: CspRaw,
 }
 
+/// Defines CSP report sent through the [Reporting API](https://developer.mozilla.org/en-US/docs/Web/API/Reporting_API).
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct CspViolationRaw {
+    #[serde(rename = "type")]
+    ty: String,
+    body: CspRaw,
+
+    /// For forward compatibility.
+    #[serde(flatten)]
+    other: BTreeMap<String, serde_json::Value>,
+}
+
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+#[serde(untagged)]
+enum CspVariant {
+    Csp(CspReportRaw),
+    CspViolation(CspViolationRaw),
+}
```

Comment:
```suggestion
enum CspVariant {
    Csp { #[serde(rename = "csp-report")] csp_report: CspRaw },
    CspViolation { body: CspRaw }
}
```

All other types don't add value, the `CspViolationRaw` just contains additional unused fields and the `CspReportRaw` is also just a deserialization helper and never used.